File: Php/app/Homegame/Listing/HomegameListingModel.php
namespace app\Homegame\Listing{

	use core\PageModel;
	use Domain\Classes\User;

	class HomegameListingModel extends PageModel {

		public $homegameModels;

		public function __construct(User $user, array $homegames){
			parent::__construct($user);
			$this->homegameModels = $this->getHomegameModels($homegames);
		}

		private function getHomegameModels(array $homegames){
			$models = array();
			foreach($homegames as $homegame){
				$models[] = new HomegameItemModel($homegame);
			}
			return $models;
		}

	}

}

File: Web/Models/HomegameListingModel.cs
using System.Collections.Generic;
using System.Linq;
using Core.Classes;
using Web.Models.HomegameModels.Listing;

namespace Web.Models{

	public class HomegameListingModel : PageModel {

	    public IList<HomegameItemModel> HomegameModels { get; set; }

	    public HomegameListingModel(User user, IEnumerable<Homegame> homegames) : base(user)
	    {
	        HomegameModels = GetHomegameModels(homegames);
	    }

		private IList<HomegameItemModel> GetHomegameModels(IEnumerable<Homegame> homegames)
		{
		    return homegames.Select(homegame => new HomegameItemModel(homegame)).ToList();
		}

        public override string BrowserTitle
        {
            get
            {
                return "Homegame List";
            }
        }
	}

}